narwhals/_arrow/series.py

```diff
@@ -163,6 +163,10 @@ def sum(self) -> int:
         pc = get_pyarrow_compute()
         return pc.sum(self._native_series)  # type: ignore[no-any-return]
 
+    def drop_nulls(self) -> int:
+        pc = get_pyarrow_compute()
+        return pc.drop_null(self._native_series)
```

Comment:
you're missing `self._from_native_series` (check `abs` / `cum_sum` below in the same file)